src/h5web/providers/my-utils.ts

```diff
@@ -2,25 +2,32 @@ import { nanoid } from 'nanoid';
 import {
   HDF5Attribute,
   HDF5Dims,
+  HDF5ExternalLink,
+  HDF5LinkClass,
   HDF5NumericType,
+  HDF5ScalarShape,
   HDF5Shape,
   HDF5SimpleShape,
+  HDF5StringType,
   HDF5Type,
   MyHDF5Dataset,
   MyHDF5Datatype,
   MyHDF5Entity,
   MyHDF5EntityKind,
   MyHDF5Group,
+  MyHDF5Link,
 } from '../providers/models';
 import { makeSimpleShape, makeStrAttr } from '../providers/mock/data';
-import { NxInterpretation } from '../visualizations/nexus/models';
+import { NxInterpretation, SilxStyle } from '../visualizations/nexus/models';
 
-type Opts = Partial<Pick<MyHDF5Dataset, 'id' | 'attributes'>>;
+type EntityOpts = Partial<Pick<MyHDF5Dataset, 'id' | 'attributes'>>;
+type GroupOpts = EntityOpts & { children?: MyHDF5Entity[] };
+type AllOrNone<T> = T | { [K in keyof T]?: never };
 
 export function makeMyGroup(
   name: string,
   children: MyHDF5Entity[] = [],
-  opts: Opts = {}
+  opts: EntityOpts = {}
```

Comment:
A bit inconsistent to have `GroupOpts` in the NxGroup creation and `children` and `EntityOps` here...

Reply:
Yeah, I know, but passing the `children` in the options here led to too much nesting in `data.ts`. 😢 However, I can replace `EntityOpts` here with `Omit<GroupOpts, 'children'>` 👍